src/OpenTracing/SpanOptions.php

```diff
@@ -43,7 +43,15 @@ public static function create(array $options)
 
                 case 'tags':
                     foreach ($value as $tag => $tagValue) {
-                        $spanOptions->tags[] = Tag::create($tag, $tagValue);
+                        if ($tag !== (string) $tag) {
+                            throw InvalidSpanOption::invalidTag($tag);
+                        }
+
+                        if (is_array($tagValue) || (!is_scalar($tagValue) && method_exists($tagValue, '__toString'))) {
```

Comment:
If `$tagValue` has `__toString` you have
`true || (!false && method_exists)`
`=== true || (true && true)`
`=== true || true`
`=== true`